### PR TITLE
Tickets/dm 11120

### PR DESCRIPTION
**Pull request by afausti**
Deployment of the SQuaSH DB to kubernetes, it is meant to be used in conjunction with the squash-api microservice.

### CODE REVIEW

File: kubernetes/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    spec:
+      containers:
+      # based on mariadb and restore qadb
+      - image: mariadb:10.3
+        name: mariadb
+        env:
+        # kubectl create secret generic mysql-pass --from-file=passwd.txt
+        - name: MYSQL_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
```
Comment (jhoblitt) [inline kubernetes/deployment.yaml:22]: Neat -- I didn't know this was possible.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	    kubectl create -f $(LOCAL_VOLUME_CONFIG);\
+	else\
+	    echo "Creating a GKE persistent volume...";\
+	    kubectl create -f $(GKE_VOLUME_CONFIG);\
+	fi
+
+	kubectl delete --ignore-not-found=true PersistentVolumeClaim mysql-volume-claim
+	kubectl create -f $(PERSISTENT_VOLUME_CLAIM_CONFIG)
+
+	kubectl delete --ignore-not-found=true deployment squash-db
```
Comment (jhoblitt) [inline Makefile:41]: FYI - you can pass the .yaml to `delete`.

File: kubernetes/service.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  name: squash-db
+  labels:
+    app: squash-api
+spec:
+  ports:
+    - protocol: TCP
+      port: 3306
+  selector:
+    app: squash-api
+  type: LoadBalancer
```
Comment (jhoblitt) [inline kubernetes/service.yaml]: `LoadBalancer` allocates a routable IP and makes the service accessible from anywhere without access control.  This probably a bad thing -- especially considering that containers are not as isolated as VMs.  Things to consider are:

* making this a `ClusterIP` that is not externally accessible.
* Setting up network ACLs along the lines of https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/ (I have not tried this)
* Network policies https://kubernetes.io/docs/tasks/administer-cluster/declare-network-policy/ (I have not tried this nor tried to add support for it to a gke cluster)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,63 @@
...
+kubectl describe deployment squash-db
+``` 
+
+and the `mariadb` container logs using:
+
+```
+kubectl logs deployment/squash-db mariadb
+```
+
+## Restoring a copy of SQuaSH's production database
```
Comment (jhoblitt) [inline README.md:53]: Keep in mind that backups would be needed for a production deployment.  The container used for the production RDS instance could likely be reused -- possibly as a k8s 'job'.

Comment (afausti) [reply README.md:53]: Nice! will open a ticket for that.

File: kubernetes/service.yaml (hunk 3 above)
Comment (afausti) [reply kubernetes/service.yaml]: Good point, will review the options here.

**Review by jhoblitt — COMMENTED** [review]
I haven't had a chance to run this yet but it looks reasonable.   I feel that all k8s deployments should include a creating/destroying a namespace to prevent resources from becoming "littered" in the default ns.

I've made some more detailed inline comments.  Essentially, access control and/or making this not externally accessible and automated backs are needed to make this deployment "production ready".

**Comment by afausti** [comment]
Changed squash-db service type to ClusterIP and opened tickets to implement the other suggestions.

**Pull request merged**